client/jetbrains/src/main/java/com/sourcegraph/cody/chat/MessageContentCreatorFromMarkdownNodes.java

```diff
@@ -0,0 +1,242 @@
+package com.sourcegraph.cody.chat;
+
+import com.intellij.ide.highlighter.HighlighterFactory;
+import com.intellij.lang.Language;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.EditorFactory;
+import com.intellij.openapi.editor.EditorSettings;
+import com.intellij.openapi.editor.colors.EditorColorsManager;
+import com.intellij.openapi.editor.ex.EditorEx;
+import com.intellij.openapi.editor.highlighter.EditorHighlighter;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.fileTypes.FileTypeManager;
+import com.intellij.openapi.fileTypes.PlainTextFileType;
+import com.intellij.ui.BrowserHyperlinkListener;
+import com.intellij.ui.ColorUtil;
+import com.intellij.util.ui.JBInsets;
+import com.intellij.util.ui.UIUtil;
+import java.awt.*;
+import java.util.Optional;
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import org.commonmark.node.*;
+import org.commonmark.node.Image;
+import org.commonmark.renderer.html.HtmlRenderer;
+import org.jetbrains.annotations.NotNull;
+
+public class MessageContentCreatorFromMarkdownNodes extends AbstractVisitor {
```

Comment:
nitpick: we might want a better name, its purpose is hard to deduce without looking at the code IMO

Smth like: 
```suggestion
public class MarkdownCodeBlockToCodeEditorVisitor extends AbstractVisitor {
```
(probably not much better, just indicating what I mean)

Reply:
Yeah, I couldn't came up with a better name :D But I will try to do something with it in the next pr's